### PR TITLE
New THREE.AudioStream

### PR DESCRIPTION
**Pull request by Mugen87**
This PR adds the possibility to use audio streams based on [HTMLAudioElement](https://developer.mozilla.org/de/docs/Web/API/HTMLAudioElement).

You can check out and debug the corresponding (simple) example right **[here](https://yume.human-interactive.org/examples/audio/misc_sound_stream.html)**. It uses `THREE.AudioAnalyser`to visualize the data of the stream.

`THREE.AudioStream` integrates nicely in the existing audio API.

```javascript
var listener = new THREE.AudioListener();
camera.add( listener );

var audioStream = new THREE.AudioStream( 'mysong.ogg', listener );
var analyser = new THREE.AudioAnalyser( audioStream );

audioStream.play();
```

### CODE REVIEW

**Comment by Mugen87** [comment]
Too bad. No emoticons in PR titles 😞

**Review by EvilPeanut — APPROVED** [review]

File: src/audio/AudioStream.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+/**
+ * @author Mugen87 / https://github.com/Mugen87
+ */
+
+function AudioStream( path, listener ) {
```
Comment (mrdoob) [inline src/audio/AudioStream.js]: I think I would follow the other classes and keep this as `function AudioStream( listener )`. And add a `setAudioElement( new Audio( path ) )`.

Comment (Mugen87) [inline src/audio/AudioStream.js]: I've tried to address your feedback in the latest commit. I removed `path` from the constructor and added a new method `setAudioElement()`.

**Comment by Mugen87** [comment]
@mrdoob Do you think we can merge this PR? Or are there (maybe conceptual) open issues?

File: src/audio/AudioStream.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+Object.assign( AudioStream.prototype, {
+
+	getOutput: function () {
+
+		return this.gain;
+
+	},
+
+	setAudioElement: function ( audioElement ) {
```
Comment (mrdoob) [inline src/audio/AudioStream.js]: I'm probably missing something but... Wouldn't it be simpler to just add this method to [Audio](https://github.com/mrdoob/three.js/blob/dev/src/audio/Audio.js)?

Comment (Mugen87) [inline src/audio/AudioStream.js]: The problem is that methods like `.play()`, `.pause()`, `.getCurrentTime()` or `.getLoop()` only work if they operate on `. audioElement` (the actual `HTMLAudioElement`). The methods of `Audio` seems to work exclusively with  `AudioBufferSourceNode`.

Comment (Mugen87) [inline src/audio/AudioStream.js]: To be more precise: `.createMediaElementSource()` returns an object of type `MediaElementAudioSourceNode`. I'm not 100% sure about that but this type of node is something different than `AudioBufferSourceNode`.

https://developer.mozilla.org/de/docs/Web/API/MediaElementAudioSourceNode

**Pull request closed without merging**